lib/dolly/collection.rb

```diff
@@ -1,19 +1,14 @@
 module Dolly
-  class Collection
-    extend Forwardable
+  class Collection < DelegateClass(Set)
     attr_accessor :rows
     attr_writer :json, :docs_class
 
-    def_delegators :@set, :clear, :empty?, :length, :+, :-
-
     def initialize str, docs_class
-      @set = Set.new
       @docs_class = docs_class
       @json = str
-    end
-
-    def first
-      to_a.first
+      initial = []
+      super(initial)
+      load
```

Comment:
no actually. that was what my problem was doing this for the last 2 hours. Without doing super, the set is never initialized, and during the `load` method at some point `self << document` is called, which cant happen if it isn't initialized